hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/statemachine/commandhandler/CloseContainerCommandHandler.java

```diff
@@ -69,73 +85,76 @@ public CloseContainerCommandHandler() {
-    final ContainerController controller = ozoneContainer.getController();
-    final long containerId = closeCommand.getContainerID();
-    LOG.debug("Processing Close Container command container #{}",
-        containerId);
-    try {
-      final Container container = controller.getContainer(containerId);
-
-      if (container == null) {
-        LOG.error("Container #{} does not exist in datanode. "
-            + "Container close failed.", containerId);
-        return;
-      }
-
-      // move the container to CLOSING if in OPEN state
-      controller.markContainerForClose(containerId);
-
-      switch (container.getContainerState()) {
-      case OPEN:
-      case CLOSING:
-        // If the container is part of open pipeline, close it via write channel
-        if (ozoneContainer.getWriteChannel()
-            .isExist(closeCommand.getPipelineID())) {
-          ContainerCommandRequestProto request =
-              getContainerCommandRequestProto(datanodeDetails,
-                  closeCommand.getContainerID(),
-                  command.getEncodedToken());
-          ozoneContainer.getWriteChannel()
-              .submitRequest(request, closeCommand.getPipelineID());
-        } else if (closeCommand.getForce()) {
-          // Non-RATIS containers should have the force close flag set, so they
-          // are moved to CLOSED immediately rather than going to quasi-closed.
-          controller.closeContainer(containerId);
-        } else {
-          controller.quasiCloseContainer(containerId,
-              "Ratis pipeline does not exist");
-          LOG.info("Marking Container {} quasi closed", containerId);
+    queuedCount.incrementAndGet();
+    CompletableFuture.runAsync(() -> {
+      invocationCount.incrementAndGet();
+      final long startTime = Time.monotonicNow();
+      final DatanodeDetails datanodeDetails = context.getParent()
+          .getDatanodeDetails();
+      final CloseContainerCommandProto closeCommand =
+          ((CloseContainerCommand) command).getProto();
+      final ContainerController controller = ozoneContainer.getController();
+      final long containerId = closeCommand.getContainerID();
+      LOG.debug("Processing Close Container command container #{}",
+          containerId);
+      try {
+        final Container container = controller.getContainer(containerId);
+
+        if (container == null) {
+          LOG.error("Container #{} does not exist in datanode. "
+              + "Container close failed.", containerId);
+          return;
         }
-        break;
-      case QUASI_CLOSED:
-        if (closeCommand.getForce()) {
-          controller.closeContainer(containerId);
+
+        // move the container to CLOSING if in OPEN state
+        controller.markContainerForClose(containerId);
+
+        switch (container.getContainerState()) {
+        case OPEN:
+        case CLOSING:
+          // If container is part of open pipeline, close it via write channel
```

Comment:
Nit: Removing "the" from the sentence, makes it incorrect. The sentence should read:

> If the container is part of **an** open pipeline, close it via write channel

The typo is the missing "an" if you want to fix it but removing "the" is not correct